### DISCUSSION TITLE
How to invalidate ImageSharp.Web cache?

### DISCUSSION THREAD

**Discussion by VILLAN3LL3** [SixLabors/ImageSharp.Web#326] [Q&A]
I use the `AzureBlobStorageCache` in my project for user profile pictures and would like to know how to invalidate the cache in case a user replaces their profile picture?

**Comment by VILLAN3LL3** [comment]
https://github.com/SixLabors/ImageSharp.Web/discussions/136

**Reply by JimBobSquarePants** [reply]
The linked question discusses deleting cached items. 

When you upload a new image at the same source location the middleware will automatically detect a change and reprocess, cache, and serve the new image. However this requires a request for the image which means you need to invalidate the browser cache. The best way to do that is to create and register a no-op IImageWebProcessor that recognises a value you would send with your image, generally a time stamp containing the last edited date and time. 

I’m fairly certain that the Umbraco CMS has an implementation of this you could copy

**Reply by VILLAN3LL3** [reply]
@JimBobSquarePants Thank you for your answer. In the Umbraco CMS repo there is [only one IImageWebProcessor implementation](https://github1s.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Cms.Imaging.ImageSharp/ImageProcessors/CropWebProcessor.cs) which does not contain something related to caching.
In my project the GET endpoint always returns the old image after DELETE or POST, even if I disable caching in my browser. Everything works fine if I disable the Imagesharp.Web middleware.

**Reply by JimBobSquarePants** [reply]
You're hitting the browser cache then. The middleware explicitly check for file data changes.

https://github.com/SixLabors/ImageSharp.Web/blob/67e82ab6699b5e23e7217ac405a0e190825436f2/src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs#L441-L489

**Reply by VILLAN3LL3** [reply]
I was able to isolate the problem, the cause was the browser cache or the default cache-control header. I have now solved the problem by setting the cache-control header to "public, must-revalidate, max-age=1" for the Blob when uploading to Azure Storage. With this it works correctly, but seems quite hacky to me....
